components/otelcol-sumo.cmake

```diff
@@ -8,11 +9,44 @@ macro(default_otc_linux_install)
   install_otc_binary()
 endmacro()
 
+macro(default_otc_darwin_install)
+  create_otc_components()
+  install_otc_config_directory()
+  install_otc_config_fragment_directory()
+  install_otc_state_directory()
+  install_otc_filestorage_state_directory()
+  install_otc_sumologic_yaml()
+  install_otc_darwin_hostmetrics_yaml()
+  install_otc_binary()
+  install_otc_uninstall_script()
+endmacro()
+
+macro(create_otc_components)
+  cpack_add_component_group("otelcol-sumo-group"
+    DISPLAY_NAME "OpenTelemetry Collector"
+    #DESCRIPTION "example group description"
```

Comment:
We should remove it before merge, as well as other unneeded comments

Reply:
I think I got it all other than one block that I'd like to keep as it should be uncommented once CPack adds a way to configure `hostArchitectures` for macOS packages.